### ISSUE TITLE
amp-google-maps with amp-bind integration 

### ISSUE THREAD

**Issue by p2er** [ampproject/amphtml#21772]
## Describe the new feature or change to an existing feature you'd like to see

For travel sites it is often required to show locations on a map and and reflect user selections (e.g. country, state, city) on that map to help inspire the user and find relevant places as markers on that map. The selection on that map should be made available for amp binding to show relevant information in an amp-list. 

For instance: 
1. User types in a country name into search panel
2. Map zooms into country location
3. User zooms into state on map
4. Places and teasers relevant to state load via amp-list
5. User selects location on map
6. Search panel updates to show user selection and can be submitted in combination with selected dates   

## Describe alternatives you've considered

One solution might be implementing an amp-list witch would then render the iframe. User selection in the search panel could update the amp state to generate a new amp-list data url and the list would re-render with a new iframe that shows the user selection. 

That seems to be a one way approach unfortunately, since the users selection inside the iframe can not be communicated back to the bound amp state. 

## Additional context

Check https://www.google.com/flights for comparison. Map based selection is reflected in the search panel and guides a user without specific knowledge of a country the the desired airport.

**Comment by aghassemi** [comment]
Thanks @p2er for the FR. There are two approaches here:
1- dedicated Maps component #12209 
2- two-way messaging for `amp-iframe` https://github.com/ampproject/amphtml/issues/9074 

two-way messaging is getting closer to reality, it has been blocked by inability for us to detect whether a message came from a user-interaction or not. We are working with the Chrome team to get that resolved ( https://github.com/mustaqahmed/user-activation-delegation ). Chrome has already shipped a version under experimental flag, it is just waiting for Spec approval.

**Comment by aghassemi** [comment]
deduping with #12209
